examples/code-highlighting/index.js

```diff
@@ -122,34 +154,15 @@ class CodeHighlighting extends React.Component {
   renderMark = (props, editor, next) => {
     const { children, mark, attributes } = props
 
-    switch (mark.type) {
-      case 'comment':
-        return (
-          <span {...attributes} style={{ opacity: '0.33' }}>
-            {children}
-          </span>
-        )
-      case 'keyword':
-        return (
-          <span {...attributes} style={{ fontWeight: 'bold' }}>
-            {children}
-          </span>
-        )
-      case 'tag':
-        return (
-          <span {...attributes} style={{ fontWeight: 'bold' }}>
-            {children}
-          </span>
-        )
-      case 'punctuation':
-        return (
-          <span {...attributes} style={{ opacity: '0.75' }}>
-            {children}
-          </span>
-        )
-      default:
-        return next()
+    if (CODE_SYNTAX_MARKS[mark.type]) {
```

Comment:
This would make example much harder to understand

Reply:
The previous code is not suitable for others to use. In addition, the code is obvious, I also believe that most people have considerable technical level. If you still think it's not easy to understand, it is recommended to add a few lines of comments.

Reply:
I agree that this makes it harder to understand.

Reply:
How should I put it?
I need to apply the feature of this example to my own project, but the original code is misleading. Using `switch-case-default` here is not the good idea.

Reply:
I agree with both. The code in its previous form is easier to understand and to customize. The code in its new form is out-of-the-box more useful as a drop in case but at the cost of making understanding what's going on a little more difficult.

I wonder if it's appropriate to split this into two examples:

- The way it is currently
- A new way called "Syntax Highlighting with Prism Themes"

Feels like more examples would be useful in this case. Projects like `next.js` provide a lot of examples and variations and I'm always glad for it.

Reply:
First of all, thanks for submitting this PR @CodePlayer 

It adds complexity which I don't like for a basic example but I like the idea of having an example using Prism Themes. For me personally, I'd appreciate having a separate "Syntax Highlighting with Prism Themes" example.

That said, I feel like there's some unnecessary complexity in this Prism example as well. I haven't gone through the example with a fine tooth comb but it's hard to parse on first viewing.